### ISSUE TITLE
'Could not find Browser.Navigation.Key' 

### ISSUE THREAD

**Issue by eratio08** [hmsk/vite-plugin-elm#1]
Hey there,

the following error occurs right after the first hot reload 

```shell
[vite-plugin-elm] Hot-swapping Main not possible: could not find Browser.Navigation.Key in the new app model
```

~~This seems be related to a change in Elm.~~ ~~The issue was already resolved in `elm-hot` https://github.com/klazuka/elm-hot/commit/1454183fce3aca9923be6fed02b076f1a1b9214e~~

Thank you for this plugin!

**Comment by hmsk** [comment]
Thanks for the report! I will look into it. It would be great if you provide a minimum reproducible repo 🙏

**Comment by eratio08** [comment]
Using https://github.com/deathman92/vite-elm-template with `elm ^0.19.1-5` and the following `Main.elm` will produce the error on save

```Elm
module Main exposing (..)

import Browser
import Html exposing (Html, div, h1, img, text)
import Html.Attributes exposing (src)



---- MODEL ----


type alias Model =
    {}


init : ( Model, Cmd Msg )
init =
    ( {}, Cmd.none )



---- UPDATE ----


type Msg
    = NoOp


update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    ( model, Cmd.none )



---- VIEW ----


view : Model -> Html Msg
view model =
    div []
        [ img [ src "/logo.svg" ] []
        , h1 [] [ text "Your Elm App is working!" ]
        ]



---- PROGRAM ----


main : Program () Model Msg
main =
    Browser.application
        { view = \model -> { title = "Test", body = [ view model ] }
        , init = \_ _ _ -> ( {}, Cmd.none )
        , update = \_ model -> ( model, Cmd.none )
        , subscriptions = always Sub.none
        , onUrlRequest = \_ -> NoOp
        , onUrlChange = \_ -> NoOp
        }

```

**Comment by eratio08** [comment]
Looks like the `elm-hot-nav-key` property is never set by the plugin as an injection point like in https://github.com/klazuka/elm-hot/blob/1454183fce3aca9923be6fed02b076f1a1b9214e/src/inject.js. This could be the main reason for not finding it.

**Comment by hmsk** [comment]
Thanks, I just could reproduce the same error. and as you pointed, `inject.js` is an exact missing piece.

Re: depending file's HMR, I just remember it's not working as expected after vite 2. will try to fix it separately.

**Comment by hmsk** [comment]
Still having trouble with the code you gave, but released a new version with fixing crashing problems which happened with HMR + `Browser.application`.
https://github.com/hmsk/vite-plugin-elm/releases/tag/v2.0.1

**Comment by alissonfpmorais** [comment]
Hello guys, hope you're doing good!

First, thanks @hmsk for the amazing work in this plugin :+1: 

I'm having the same issue reported by @eratio08 and I've tried to work on this, but I don't have much knowledge to understand the impact of the changes that I'm doing.
What I've done so far is this update https://github.com/alissonfpmorais/vite-plugin-elm/blob/fix/hmr_and_navkey/src/index.ts#L248
The error has gone, but it doesn't mean that everything is working properly. Is this right? How can I properly test if everything is fine?
Let me know if I can help in anything else!

**Comment by hmsk** [comment]
Thanks, @alissonfpmorais, that sounds awesome! let me take a look into that with some of my samples which had the same errors. Anyways, it'd be great if you send out a PR to merge easily later. I'd love to have your name on contributors at least.

**Comment by alissonfpmorais** [comment]
Thanks @hmsk for the quick answer!
I've created the PR, let me hear from you your thoughts after testing against your samples.

**Comment by hmsk** [comment]
I found out the root cause of this error with the sample code that HMR codes were trying to replace the state of `Browser.Navigation` but it doesn't exist at all. Fixed in `2.1.2` https://github.com/hmsk/vite-plugin-elm/releases/tag/v2.1.2
